File: LeetCode/28.find-the-index-of-the-first-occurrence-in-a-string.cpp
/*
 * @lc app=leetcode id=28 lang=cpp
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
class Solution {
public:
    int strStr(string s, string x) {

        int n=s.length();
        int m=x.length();

        char ch=x[0];

        int i=0;
        int j=0;


        while(i<n)
        {
            if(s[i]==ch)
            {
                if(s.substr(i,m)==x)
                {
                    return i;
                }
                else
                {
                    i++;
                }
            }
            
            else
            {
                i++;
            }

        }


        return -1;


       



       
      
    }
};
// @lc code=end

File: LeetCode/28.find-the-index-of-the-first-occurrence-in-a-string.java
/*
 * @lc app=leetcode id=28 lang=java
 *
 * [28] Find the Index of the First Occurrence in a String
 */

// @lc code=start
class Solution {
    public int strStr(String s1, String s2) {

        int ind = -1;

        for (int i = 0; i < s1.length(); i++) {
            if (s1.charAt(i) == s2.charAt(0)) {
                ind = i;
                int j = 0;
                while (i < s1.length() && j < s2.length() && s1.charAt(i) == s2.charAt(j)) {
                    i++;
                    j++   ;
                } 

                if (j == s2.length()) {
                    return ind;
                } else {
                    i = ind;
                    ind = -1;
                }

            }
        }

        return ind;

    }
}
// @lc code=end
